### DISCUSSION TITLE
Prometheus: Alternative values for exemplars plotted in rate

### DISCUSSION THREAD

**Discussion by gtk-grafana** [grafana/grafana#60194] [Observability]
It appears as though Exemplars are given a default value of 1 for counters, which is not relevant when visualizing the exemplars.
How can we place exemplars on the graph in such situations that their placement is useful to end users, instead of giving them all the same value and plotting all exemplars on the same y-axis value?

<img width="1706" alt="image" src="https://user-images.githubusercontent.com/109082771/207139385-0c222655-ebc6-4d58-83f9-97a6e7c22614.png">

**Comment by gtk-grafana** [comment]
Idea: in the exemplar modal, allow users to click on each property of the exemplar to plot the exemplar based on the value of that prop?

<img width="396" alt="image" src="https://user-images.githubusercontent.com/109082771/207130990-d7e2471d-bdb7-41d0-8e03-4c71b408252c.png">

But most of these "values" are actually strings, and won't give us a value for the y-axis that is useful. So is there anything else we can use besides the value to plot? 

Are exemplars for counters doomed to always plot on the same y-axis position?

**Reply by gtk-grafana** [reply]
Aka plotting the exemplars on an additional vertical axis grouped by the string values, which is still going to lead to groups of exemplars sharing the same vertical position, but it would impart some additional information to the end-user.

**Comment by gtk-grafana** [comment]
Would plotting the exemplars above or below the highest/lowest point on the Y axis be a potential solution?

**Reply by gtk-grafana** [reply]
This has been suggested as a potential solution by a customer using Exemplars, and would be a pretty low effort implementation. CC: @itsmylife @grafana/observability-metrics

**Reply by itsmylife** [reply]
I wonder how good/bad from the point of UX.

**Comment by gtk-grafana** [comment]
I fail to understand the question, what other value can we use for a count besides 1? They are log entries associated with a count of some metric, for example, the total number of HTTP requests made to a server, each exemplar is just the log entry of the HTTP request that incremented the count.

**Reply by gtk-grafana** [reply]
The only solution I can think of for this use case to is allow the exemplars to be broken out into a table view.

So here's a few new questions: 
1. Is there any value visualizing exemplars of a count if they're always going to be grouped together where the x-axis is 1? Should we hide exemplars in this use-case (provided there is another solution to add them to dashboards)?
2. It doesn't look like you can currently filter out the exemplars in the transformations to create a tabular view of the exemplars. Should we be supporting a transformation that allows people to pull exemplars (that might not be visualized) from a dashboard source or raw query?

**Comment by leeoniya** [comment]
we could potentially offer some way in the panel options to choose which exemplar field to use for Y. if that field is non-numeric it could look something like the string/enum axis in https://github.com/grafana/grafana/pull/60068 but without the connected stepped lines, just the exemplar points). or to omit exemplars altogether and re-use them in another table panel with a `Dashboard` datasource.

![image](https://user-images.githubusercontent.com/43234/206635953-ce2273f8-f9ae-4d47-8456-a46164a2d787.png)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
